voidcrew/code/game/objects/structures/beach/seacrystal.dm

```diff
@@ -0,0 +1,104 @@
+/obj/structure/spawner/sea_crystal
+	name = "sea crystal"
+	desc = "A large crystal. Terrible monsters are pouring out from all around it."
+	icon = 'voidcrew/icons/obj/seacrystal.dmi'
+	icon_state = "seacrystal"
+	faction = list("beach")
+	max_mobs = 2
+	spawn_time = 10
+	max_integrity = 1350
+	pixel_x = -18
+	pixel_y = -5
+	mob_types = list(/mob/living/simple_animal/hostile/carp/megacarp/beach)
+	move_resist = INFINITY
+	anchored = TRUE
+	resistance_flags = FIRE_PROOF | LAVA_PROOF
+	layer = FLY_LAYER
+	light_color = LIGHT_COLOR_ELECTRIC_GREEN
+	light_power = 1
+	light_range = 4
+	var/cooldown_time = 20 SECONDS
+	var/newcolor = "#1302ad"
+
+	COOLDOWN_DECLARE(summon_cooldown)
+
+/obj/structure/spawner/sea_crystal/Initialize()
+	. = ..()
+	for(var/turf in RANGE_TURFS(1, src))
+		if(ismineralturf(turf))
+			var/turf/closed/mineral/mineral = turf
+			mineral.ScrapeAway(null, CHANGETURF_IGNORE_AIR)
+
+/obj/structure/spawner/sea_crystal/deconstruct(disassembled)
+	playsound(loc,'sound/effects/tendril_destroyed.ogg', 200, FALSE, 50, TRUE, TRUE)
+	new /obj/effect/temp_visual/seacrystal/sparks(loc)
+	new /obj/item/sea_crystal(loc)
+	return ..()
+
+/obj/structure/spawner/sea_crystal/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	summon_minions()
+
+/obj/structure/spawner/sea_crystal/attack_animal(mob/living/simple_animal)
+	. = ..()
+	summon_minions()
+
+/obj/structure/spawner/sea_crystal/bullet_act(obj/projectile/proj)
+	. = ..()
+	summon_minions()
+
+/obj/structure/spawner/sea_crystal/proc/summon_minions()
+	if(!COOLDOWN_FINISHED(src, summon_cooldown))
+		log_runtime("cooldown not over")
```

Comment:
I don't think you need to runtime this, it's not realy an error, just return quietly

Reply:
Was doing that for testing, will remove tomorrow